### DISCUSSION TITLE
Loss of text color with certain (large) images when adding text

### DISCUSSION THREAD

**Discussion by inpresif** [ImageMagick/ImageMagick#3572] [Development] [answered]
We noticed that sometimes our images are not processed correctly and I am trying to wrap my head around it.
Could be a filesize issue, or even a meta info issue. Not sure. For example: I noticed that people who use canvas have some issues at times that the image cannot be read.

Last case used this image:
https://manybot-thumbnails.s3.eu-central-1.amazonaws.com/fb101641712035296/ca/big_d5225f386e151185deaaecfde0cd3587.jpg

Obviously a huge image so after we suggested they compress it with tinypng it worked fine again.

This is the code that broke the textcolor. Yellow became blue, White > grey and some other weird results.

`/usr/bin/magick "https://s3.eu-central-1.amazonaws.com/ca151185deaaecfde0cd3587.jpg" -resize 900 \( -size 200x200 -background none -fill "Yellow" -font "./lib/handb.ttf" -gravity Center label:"John" -trim \) -quality 40 -composite "3884084144974854.jpg"`

Any idea how to prevent this, or what the reason could be? Even after running this code the filesize still was 650Kb, whereas after precompressing it, it's only 45Kb.

Could/can we add an extra command to compress it before running the text effects on it? It figured -resize was sufficient but apparently it is not. Tried using -thumbnail instead of -resize but that did not make difference either.

Additionally, I see the original is 300dpi / 32 bit depth, the compressed, working one is 96dpi, 24 bit depth

Thanks.

**Comment by fmw42** [comment, marked as answer]
Your image is CMYK (with  U.S. Web Coated (SWOP) v2 profile) and you are trying to put an RGB colored text onto it. You should convert your image to sRGB before adding text. You can do that with -colorspace sRGB or -profile .../sRGB.icc

**Reply by inpresif** [reply]
Thanks Fred, a lifesaver as always :)
(you helped me lots of times on the old forum (another username for me there))

**Reply by inpresif** [reply]
I was also looking into this thread (https://legacy.imagemagick.org/discourse-server/viewtopic.php?t=18241 )where you suggested 
`-units PixelsPerInch rose1.png -resample 300`
to increase DPI, could I do that as well for - in reverse to 72) best results? Or does your sRGB already optimize perfectly?

**Reply by fmw42** [reply]
I do not understand. That command has nothing to do with CMYK vs RGB.  If this is a different question, please post a new question or issue.

**Reply by inpresif** [reply]
Gotcha. Already ran the command with your suggestion, not only does it work well (it does however darken the image slightly), it decreased filesize from 650kb to 60kb. All good.

**Reply by fmw42** [reply]
Did you use -colorspace or -profile.  profiles would be a bit better color quality, but harder to use for all cases depending upon whether you have an input profile or not. If no profile, then you need to assign a profile before converting to sRGB.

**Reply by inpresif** [reply]
I used -colorspace sRGB. It was easy.
Actually the lighter base image was a previously compressed one via tinypng. The original had the same darkness so all is perfect.

**Comment by snibgo** [comment]
The input file is encoded as CMYK, with an embedded ICC profile. If you want to convert to sRGB, I suggest using "-profile", not "-colorspace", for accuracy.

Note that named colours such as "yellow" are really shorthand for numbers such as "#ff0" where the numbers are assumed to represent RGB, not CMYK.

**Reply by inpresif** [reply]
Thanks snibgo, it's all sorted.
I might try the other (ICC) way as well as a test foir good measure.
